src/Microsoft.ML.Data/Transforms/ValueMapping.cs

```diff
@@ -396,29 +414,32 @@ internal sealed class Options
         }
 
         internal ValueMappingTransformer(IHostEnvironment env, IDataView lookupMap,
-            string keyColumn, string valueColumn, (string outputColumnName, string inputColumnName)[] columns)
+            DataViewSchema.Column lookupKeyColumn, DataViewSchema.Column lookupValueColumn, (string outputColumnName, string inputColumnName)[] columns)
             : base(Contracts.CheckRef(env, nameof(env)).Register(nameof(ValueMappingTransformer)), columns)
         {
-            Host.CheckNonEmpty(keyColumn, nameof(keyColumn), "A key column must be specified when passing in an IDataView for the value mapping");
```

Comment:
> Host.CheckNonEmpty [](start = 11, length = 19)

maybe check lookupKeyColumn.HasValue

Reply:
We can't. This is a struct, not a class.

---
In reply to: [271577590](https://github.com/dotnet/machinelearning/pull/3180#discussion_r271577590) [](ancestors = 271577590)